Modules/socketmodule.c

```diff
@@ -5514,16 +5514,22 @@ socket_ntohl(PyObject *self, PyObject *arg)
 
     if (PyLong_Check(arg)) {
         x = PyLong_AsUnsignedLong(arg);
-        if (x == (unsigned long) -1 && PyErr_Occurred())
+        if (x == (unsigned long) -1 && PyErr_Occurred()) {
+            assert(PyErr_ExceptionMatches(PyExc_OverflowError));
+            PyErr_SetString(PyExc_OverflowError,
+                            "ntohl: Python int does not fit in C 32-bit "
+                            "unsigned integer");
             return NULL;
+        }
 #if SIZEOF_LONG > 4
         {
             unsigned long y;
             /* only want the trailing 32 bits */
             y = x & 0xFFFFFFFFUL;
             if (y ^ x)
                 return PyErr_Format(PyExc_OverflowError,
-                            "int larger than 32 bits");
+                                    "ntohl: Python int too large to convert "
```

Comment:
indeed. I think missed it because there was a silently truncation in htons and in ntohs until bpo-28332.